lxd/instance/drivers/driver_qemu.go

```diff
@@ -1383,23 +1387,23 @@ func (d *qemu) start(stateful bool, op *operationlock.InstanceOperation) error {
 	revertFunc, unixListener, err := device.DiskVMVirtiofsdStart(d.state.OS.ExecPath, d, configSockPath, configPIDPath, "", configMntPath, nil)
 	if err != nil {
 		var errUnsupported device.UnsupportedError
-		if errors.As(err, &errUnsupported) {
-			d.logger.Warn("Unable to use virtio-fs for config drive, using 9p as a fallback", logger.Ctx{"err": errUnsupported})
-
-			if errUnsupported == device.ErrMissingVirtiofsd {
-				// Create a warning if virtiofsd is missing.
-				_ = d.state.DB.Cluster.UpsertWarning(d.node, d.project.Name, dbCluster.TypeInstance, d.ID(), warningtype.MissingVirtiofsd, "Using 9p as a fallback")
-			} else {
-				// Resolve previous warning.
-				_ = warnings.ResolveWarningsByNodeAndProjectAndType(d.state.DB.Cluster, d.node, d.project.Name, warningtype.MissingVirtiofsd)
-			}
-		} else {
+		if !errors.As(err, &errUnsupported) {
```

Comment:
im not enjoying this alternative version because we now have 2 duplicated calls to warnings.ResolveWarningsByNodeAndProjectAndType()

What was the linter error you were resoling here?

Reply:
It was something like : 
```
Instead of using:

if c { <if-logic> } else { <else-logic> return }

Use:

if !c { <else-logic> return }
<if-logic>
```

Reply:
ah ok, I think I missed that there were already 2 calls to  warnings.ResolveWarningsByNodeAndProjectAndType previously. This one is ok then. Thanks